scripts/utils/Git.ts

```diff
@@ -301,34 +299,40 @@ class Git {
         }
 
         try {
-            console.log(`🔄 Fetching missing ref: ${ref}`);
-            await exec(`git fetch --no-tags --depth=1 ${remote} ${ref}`, {
+            log(`🔄 Fetching missing ref: ${ref}`);
+            await exec(`git fetch ${remote} ${ref} --no-tags --depth=1 --quiet`, {
                 encoding: 'utf8',
                 cwd: process.cwd(),
             });
 
             // Verify the ref is now available
             if (!this.isValidRef(ref)) {
-                throw new Error(`Reference ${ref} is still not valid after fetching`);
+                throw new Error(`Reference ${ref} is still not valid after fetching from remote ${remote}`);
             }
         } catch (error) {
             throw new Error(`Failed to fetch git reference ${ref}: ${error instanceof Error ? error.message : String(error)}`);
         }
     }
 
-    static getMainBaseCommitHash(remote: string): string {
-        // Fetch the main branch from the specified remote to ensure it's available
-        try {
-            execSync(`git fetch ${remote} main --no-tags --depth=1 -q`, {encoding: 'utf8'});
-        } catch (error) {
-            throw new Error(GIT_ERRORS.FAILED_TO_FETCH_FROM_REMOTE);
+    static async getMainBranchCommitHash(remote?: string): Promise<string> {
+        const baseRefName = IS_CI ? (GITHUB_BASE_REF ?? GITHUB_CONSTANTS.DEFAULT_BASE_REF) : GITHUB_CONSTANTS.DEFAULT_BASE_REF;
+
+        // Fetch the main branch from the specified remote (or locally) to ensure it's available
+        if (IS_CI || remote) {
+            await exec(`git fetch ${remote ?? 'origin'} ${baseRefName} --no-tags --depth=1 -q`, {encoding: 'utf8'});
```

Comment:
I'd recommend removing `-q` for better visibility:


```suggestion
            await exec(`git fetch ${remote ?? 'origin'} ${baseRefName} --no-tags --depth=1`, {encoding: 'utf8'});
```